syntax = "proto3";

package unitystreamer;

service UnityStreamer{
  rpc StreamData(stream UnityData) returns (Received){}
}

message UnityData {
  int32 width = 1;
  int32 height = 2;
  bytes image_data = 3;
  bytes depth_data = 4;
  int64 timestamp = 5;
  repeated int32 params =6;
}

message Received{
  int64 timestamp = 1;
}
